var/spack/repos/builtin/packages/unifycr/package.py

```diff
@@ -67,8 +77,17 @@ def hdf5_compiler_path(name):
                                          lambda x: spec['numactl'].prefix))
         args.extend(self.with_or_without('hdf5', hdf5_compiler_path))
 
-        if '+debug' in spec:
-            args.append('--enable-debug')
+        if '+fortran' in spec:
+            args.append('--enable-fortran')
+
+        if '+pmpi' in spec:
+            args.append('--enable-mpi-mount')
+
+        if '+pmi' in spec:
+            args.append('--enable-pmi')
+
+        if '+pmix' in spec:
+            args.append('--enable-pmix')
 
         if spack.config.get('config:debug'):
```

Comment:
@becker33 I agree that would make more sense. I originally set this up when recommended [here](https://github.com/spack/spack/pull/9594#discussion_r226810586), however we got rid of our `+debug` variant as well and now to get a debug build, the proper `CFLAGS` need to be set.

I could get this to work by adding `flag_handler = build_system_flags` and changing line 92 to
```
if 'cflags="-g -O0"' in spec:
```
 but that feels wrong. Is there a way in my package to check what cflags the user has passed in and then I can enable this option if `-g` and `-O0` are there?

Alternatively, this build option isn't exactly necessary and I could just remove these lines.

Reply:
Does `unifycr cflags="-g -O0" only build properly with debug symbols when `flag_handler = build_system_flags`? If so, you should put that in the package.

You could also modify the flag_handler to check the flags before passing them on.

```
debug_build = False

def flag_handler(self, name, flags):
    if name in ('cflags', 'cppflags'):  # pre-processor flags also get applied to C code
        if '-g' in flags:
            self.debug_build = True
    return (None, None, flags)
```
That code is exactly the same as `flag_handler = build_system_flags` except that it also updates the `debug_build` variable to use later.

Reply:
> Does `unifycr cflags="-g -O0" only build properly with debug symbols when `flag_handler = build_system_flags`? If so, you should put that in the package.

Yes.

> That code is exactly the same as `flag_handler = build_system_flags` except that it also updates the `debug_build` variable to use later.

Great suggestion! I modified the flag_handler to check the flags and set a `debug_build` variable.